### PR TITLE
Update to the contributing.md file

### PR DESCRIPTION
**Pull request by Chizobaonorh**
I made significant changes and rewrote the contributing.md file for easy comprehension, and also added graphic images for easy navigation. @gbowne1  @LOGESH-B  @BlackBond06

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,79 +1,114 @@
...
-let us know that way issues do not go stale.
+Anyone with basic programming skills down to grammer skills can contribute immensely to this project.
+Areas requiring contribution includes:
 
-Some tasks may not be beginner friendly even if tagged `good first issue`.. so try and judge the task accordingly.
-If you are a beginner, there are smaller tasks a beginner can work on such as style issues.
+- **Error checks**: Identify any mispellings or errors in documentation? 
 
-You should already be familiar with React 17 and or 18, JavaScript ES5/ES6/ES7.
+- **Documentation**: Possess strong documentation skills, and believe you can help elevate this projects' documentation even better?
```
Comment (gbowne1) [inline CONTRIBUTING.md:12]: There are no strict requirements for documentation skills, good spelling, punctuation and grammer or using a spelling and grammar checker tool are all that are required, other than knowlege of  the  Markdown format.

**Review by BlackBond06 — COMMENTED** [review]
I would also suggest you use easy and simple to understand words.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,79 +1,114 @@
...
+
+## Select an issue
+
+If you are not sure what to work on, review the issues list first. There are also TODO's listed in the included [TODO.md](./TODO.md) file. 
+
+@gbowne1 and other project maintainers will assign users to issues on a first come, first serve basis. If you would like to work on an issue, feel free to indicate by tagging the mentioned persons.
+
+## Create an issue
+
+If you notice a typo, slight bug in the code and there are no lodged complaints on this yet, select the issues icon at the top, and on the right side of the next page select new issues and lodge your complain.
```
Comment (gbowne1) [inline CONTRIBUTING.md:28]: This could be easier to write if we change `If you notice a typo, slight bug in the code and there are no lodged complaints on this yet, select the issues icon at the top, and on the right side of the next page select new issues and lodge your complain.` 

to:

`Have you noticed something that did not look or work right in our applicaton, or a typographical error, wrong syntax, grammar, spelling or other issue?  

- Check to see if an issue has been created for whatever you have found  in <https://github.com/gbowne1/codebooker/issues>
- If there is not a issue that has already been created, Click on the green [New Issue] button.
- Describe your issue as well as you can:
  - include code snippits where the issue
  - include screen shots
  - include a short video
  
  If the issue appears to be a browser, operating system, device specific issue, let us know what those are.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,79 +1,114 @@
...
 
-I have not created an official style guide yet. I also have not created a Figma or layout. If anyone would like to create a style guide for us. Create a Discussion. I have typically let the community help
+## Fork this repository
 
-## Development
+After choosing what to work on, move to fork and setup your own repository.
 
-If you are not sure what to work on, review the issues list. There are also TODO's that will listed in the included [TODO.md](./TODO.md) file.
+On the project's repository page, look to the right side of the page to locate a fork tab, click on this icon, and this would redirect you to a new page with further instructions.
```
Comment (gbowne1) [inline CONTRIBUTING.md:34]: This might be a great place in the future to provide instructions for both cloning and forking.  They are similar process, but work differently especially when it comes to version control.  "Fork" was meant to be used for a derivitive project, and also meant for developers who wished to make their own version of a project, or a developer could use parts of a project to create their own project within the terms of the original project's license of course..   "Clone" is just that, a `clone` of the project, not meant to be used for creating any derivitive project that would be different from the main repository.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,79 +1,114 @@
...
+
+![](./docs/ForkedBranch.png)
+
+## Editor & IDE
 
-The browser also gives us things like localStorage, IndexedDB, Session Storage, Cookies and Cache Storage. Please make use of them for
+The repository contains folders with project appropriate settings and configurations for Visual Studio code, but beyond that, we are tool and editor/IDE agnostic so you can use whatever editor or IDE or you like.
 
-Please also make sure you use accessibility (A11y) with themes and styles paying attention to focus styling, contrast and keyboard-accesibility.
+Follow the instructions in [DEVSETUP](/docs/DEVSETUP.md) to setup your IDE workspace.
```
Comment (gbowne1) [inline CONTRIBUTING.md:54]: DEVSETUP.md will be a file used for:

Instructions for developers, programmers, coders to:

Clone or Fork the project
Set up the project in their IDE or editor
Set up their environment to work in the project, including any environment variables, yarn/npm/pnpm package(s), editor extensions or plugins needed
Setting up the connection to the database
Setting up the local development database, MongoDB

Basically any specific instructions a beginner or even an advanced developer/programmer needs to know to get started working in the project.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,79 +1,114 @@
...
 
-### PR's
+## Prerequisites to use Database for programmers
```
Comment (gbowne1) [inline CONTRIBUTING.md:56]: This section doesn't appear to be using text for using the database.

The original heading I had ### PR's was intended to create short instructions on how to create and submit Pull Requests (PR's), this appears now at line 70

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,79 +1,114 @@
...
+Adequate knowledge and familiarity with React 17 and or 18, JavaScript ES5/ES6/ES7.
+
+React gives us built-in hooks such as useEffect, useSelector, useRef, useHook, useCallback, useContext, useMemo, useForm, useHistory, etc. Ensure to make the most of these provided React built-ins.
+
+Also, use accessibility (A11y) with themes and styles paying attention to focus styling, contrast and keyboard-accesibility.
+
+Lastly, the browser gives us things like localStorage, IndexedDB, Session Storage, Cookies and Cache Storage.
+
+
+> Note: If assigned an issue and along the line you change your mind or can't figure it out, ensure to notify the team so it can be reassigned to another. <br><br>
```
Comment (gbowne1) [inline CONTRIBUTING.md:67]: This section needs to be removed and included in the section about creating an issue.

**Review by gbowne1 — CHANGES_REQUESTED** [review]
- [x] Checked out this documentation Pull Request change locally for review

I have made some comments about the text for you to review, comment back if you wish for clarification, etc.

I am requesting some minor changes here.  However, the majority of this change looks good.  The rest is fine.

You can commit changes to your PR branch to update and I will re-review..  Thank you for the opportunity to review your PR @Chizobaonorh!  

We are currently requiring a minumim of two PR approvals to merge from @LOGESH-B and @BlackBond06.  

If you need anything else, use the Slack documentation channel.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Chizobaonorh) [reply CONTRIBUTING.md:28]: Okay this is a much better alternative, will get this fixed.

File: CONTRIBUTING.md (hunk 3 above)
Comment (Chizobaonorh) [reply CONTRIBUTING.md:34]: I felt this would make this page more choked up, i wanted to provide information to this on the devsetup page, or what do you think?

File: CONTRIBUTING.md (hunk 4 above)
Comment (Chizobaonorh) [reply CONTRIBUTING.md:54]: Exactly, this was my exact same thought process, that is why i did not expantiate on this in this particular file.

File: CONTRIBUTING.md (hunk 5 above)
Comment (Chizobaonorh) [reply CONTRIBUTING.md:56]: I felt placing it after instructions on forking this repository and setting up the IDE was ideal, since they follow suit

File: CONTRIBUTING.md (hunk 6 above)
Comment (Chizobaonorh) [reply CONTRIBUTING.md:67]: Okay i will do that.

**Comment by Chizobaonorh** [comment]
> @gbowne1 @Chizobaonorh
> 
> I can't find any steps to setup the .env for backend and instruction to run the setup.js file. It is very important to setup this project locally

I am not sure this was meant to be included in the contributing file, my best guess would be for this to be included in the devsetup file

**Comment by gbowne1** [comment]
@LOGESH-B This file CONTRIBUTING.md is basically all about how to contribute which would include only information on how to clone/fork, pull requests, making bug reports, creating issues, maybe even links to the TODO list

We should include somehow within the file, that we require a minimum of two (2) reviews to be completed before we will merge.

DEVSETUP.md can house all of the things necessary for the developer to get their environment  set up and working.

File: CONTRIBUTING.md (hunk 3 above)
Comment (gbowne1) [reply CONTRIBUTING.md:34]: Yes, @Chizobaonorh, that information for would actually be better for the DEVSETUP.md file

File: CONTRIBUTING.md (hunk 6 above)
Comment (gbowne1) [reply CONTRIBUTING.md:67]: @Chizobaonorh Ok.

File: CONTRIBUTING.md (hunk 5 above)
Comment (gbowne1) [reply CONTRIBUTING.md:56]: @Chizobaonorh I agree.

File: CONTRIBUTING.md (hunk 4 above)
Comment (gbowne1) [reply CONTRIBUTING.md:54]: @Chizobaonorh Good. Hopefully we can get to the DEVSETUP.md file soon.

File: CONTRIBUTING.md (hunk 2 above)
Comment (gbowne1) [reply CONTRIBUTING.md:28]: @Chizobaonorh Ok. Let me know when you have updates.

**Review by gbowne1 — APPROVED** [review]
The changes to this PR look good.  

Some things ideally could be summarized, though.

Approving for merge.

**Comment by Chizobaonorh** [comment]
> The changes to this PR look good.
> 
> Some things ideally could be summarized, though.
> 
> Approving for merge.

Can you point the things you think i can summarize?

**Comment by gbowne1** [comment]
Lets wait for that till we finish what we have.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,79 +1,133 @@
...
 
-You should already be familiar with React 17 and or 18, JavaScript ES5/ES6/ES7.
+- **Documentation**: Possess good documentation skills: Good spelling, punctuation, knowledge of Markdown and grammar checker tool, and you believe you can help elevate this projects' documentation even better? 
 
-## Style
+- **Code contribution**: Identify a bug in this app or thought of a new feature that could plunge this app to a new dimension? 
+
+## How to contribute
+
+Developers often derive more convenience in seeking permission from the project to contribute, so these next headings will get you started in no time contributing.
```
Comment (LOGESH-B) [inline CONTRIBUTING.md:18]: I can't able to understand on 1st read

**Review by LOGESH-B — APPROVED** [review]
Hii @Chizobaonorh 
Thanks for your valuable time to complete this docs
Approving for merge🤝

File: CONTRIBUTING.md (hunk 7 above)
Comment (gbowne1) [reply CONTRIBUTING.md:18]: Yeah, that wording is confusing, and could probably be removed.

Comment (Chizobaonorh) [reply CONTRIBUTING.md:18]: > I can't able to understand on 1st read

Okay will work on this

Comment (Chizobaonorh) [reply CONTRIBUTING.md]: > Yeah, that wording is confusing, and could probably be removed.

Yeah i have simplified this, it reads better now.

Comment (gbowne1) [reply CONTRIBUTING.md]: I just saw. Looks much better.  The original text was placed there because I had so many people in this (and other projects too) ask me if they could please work on a issue.  I understand that is just being polite and that was fine, but it is not necessary that people ask for permission to work on an issue.  I largely prefer that people just say they would like to work on an issue, and one of our maintaners can assign.

Comment (Chizobaonorh) [reply CONTRIBUTING.md]: Yes, i totally agree with that

**Review by BlackBond06 — APPROVED** [review]
Updates made in this PR greatly improves the contribution.md file, enhancing readability and comprehension. Thanks for the opportunity to review you PR @Chizobaonorh. Approving for merge.

**Comment by gbowne1** [comment]
Alright, merging now. Thanks for your contribution @Chizobaonorh and the opportunity to review your PR.
